### PR TITLE
Switching to Turbo for management

### LINKED ISSUE #47: Consider Migrating from Lerna to Turbo
Consider migrating mono repo management to Turbo for both speed and ease of use. Also aligns with other usage of Turbo/Next.js in the organization.

### PR DESCRIPTION
**Pull request by srtfisher**
- Moves to Turbo for Monorepo management.
- Introduces a shared Typescript configuration package.
- Upgrades to Node 18

Resolves #47

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      // (and not the monorepo root)
+      "outputs": ["build/**"]
+    },
+    "test": {
+      // A package's `test` script depends on that package's
+      // own `build` script being completed first.
+      "dependsOn": ["build"],
+      // A package's `test` script should only be rerun when
+      // either a `.tsx` or `.ts` file has changed in `src` or `test` folders.
+      "inputs": ["src/**/*.tsx", "src/**/*.ts", "test/**/*.ts", "test/**/*.tsx"]
```
Comment (stevenslack) [inline turbo.json:20]: Any reason to not also check for `.js` files?

File: package.json [hunk 2]
```diff
@@ -18,6 +23,6 @@
     "eslint-plugin-jsx-a11y": "^6.6.1",
     "eslint-plugin-react": "^7.30.1",
     "eslint-plugin-react-hooks": "^4.6.0",
```
[the remark this answers is not in the document]
Comment (srtfisher) [reply package.json:25]: I'm not entirely sure why there is eslint dependencies in the root

**Comment by srtfisher** [comment]
@stevenslack I removed `eslint` from the root. Ideally we'd get an eslint package here soon enough.

**Review by stevenslack — APPROVED** [review]
Full turbo!

**Pull request merged**